### PR TITLE
HPCC-19424 Switch over to using WuAttr in code generator

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman please review.

The main change is to add entries to the enumeration for all the different attributes that can be added to the graph - which will mean they are now reported back to Gordon by the wudetails function.  I went through the code generator switching to using the enumeration in as many places as possible to ensure that I caught them all.  There are still a very few direct calls, but they are almost all code generator debugging options.  One of the enumerations WaIdAmbiguousGraph will be split in two by a subsequent change (which will also change the workunit graph format), but I preferred to keep it a separate PR.
(I have run the compiler regression suite to check for changes, and will run again on this final version. 
 It is worth ensuring smoke test passes as well.)

File: system/jlib/jstring.cpp [hunk 1]
```diff
@@ -2486,10 +2486,10 @@ StringBuffer & ncnameUnescape(char const * in, StringBuffer & out)
...
-bool startsWith(const char* src, const char* dst)
+bool startsWith(const char* src, const char* prefix)
 {
-    while (*dst && *dst == *src) { src++; dst++; }
-    return *dst==0;
+    while (*prefix && *prefix == *src) { src++; prefix++; }
+    return *prefix==0;
 }
 
 bool startsWithIgnoreCase(const char* src, const char* dst)
```
Comment (richardkchapman) [inline system/jlib/jstring.cpp]: Why not fix this one if you are going to fix that one...

**Comment by richardkchapman** [comment]
@ghalliday One trivial comment

File: system/jlib/jstring.cpp (hunk 1 above)
Comment (ghalliday) [reply system/jlib/jstring.cpp]: Energy - this was the only one I was calling.  I'll make the same change.

**Comment by ghalliday** [comment]
@richardkchapman fixed

File: system/jlib/jstring.cpp (hunk 1 above)
Comment (richardkchapman) [reply system/jlib/jstring.cpp]: I could complain about endsWith too... but I won't

**Pull request merged**